src/Expression/ForClasses/IsA.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Arkitect\Expression\ForClasses;
+
+use Arkitect\Analyzer\ClassDescription;
+use Arkitect\Expression\Description;
+use Arkitect\Expression\Expression;
+use Arkitect\Rules\Violation;
+use Arkitect\Rules\ViolationMessage;
+use Arkitect\Rules\Violations;
+
+final class IsA implements Expression
+{
+    /** @var string[] */
+    private $allowedFqcnList;
+
+    public function __construct(string ...$allowedFqcnList)
+    {
+        $this->allowedFqcnList = $allowedFqcnList;
+    }
+
+    public function describe(ClassDescription $theClass, string $because): Description
+    {
+        $allowedFqcnList = implode(', ', $this->allowedFqcnList);
+
+        return new Description("should inherit from one of: $allowedFqcnList", $because);
+    }
+
+    public function evaluate(ClassDescription $theClass, Violations $violations, string $because): void
+    {
+        if (!$this->isA($theClass, ...$this->allowedFqcnList)) {
+            $violation = Violation::create(
+                $theClass->getFQCN(),
+                ViolationMessage::selfExplanatory($this->describe($theClass, $because))
+            );
+
+            $violations->add($violation);
+        }
+    }
+
+    private function isA(ClassDescription $theClass, string ...$allowedFqcnList): bool
+    {
+        foreach ($allowedFqcnList as $allowedFqcn) {
+            if (
+                \is_a($theClass->getFQCN(), $allowedFqcn, true)
+                || \is_subclass_of($theClass->getFQCN(), $allowedFqcn)
```

Comment:
Do we really need to check both? Both do the same, except that `is_a()` also returns `true` for the class itself (like `is_a(A::class, A::class)`), AFAIK, so `is_subclass_of()` should never be called.

Reply:
> unfortunately, `is_a` doesn't check for interface implementation

I think it does: https://3v4l.org/91Yt8

AFAIK, `is_a` does the same as `instanceof` (and it is even more powerful, since it can check string class names and not only instances, if the third parameter is `true`).

Reply:
HAH! u r correct, I will fix this

Reply:
fixed and force pushed